src/main/common/maths.c

```diff
@@ -21,6 +21,34 @@
 #include "axis.h"
 #include "maths.h"
 
+// http://lolengine.net/blog/2011/12/21/better-function-approximations
+// Chebyshev http://stackoverflow.com/questions/345085/how-do-trigonometric-functions-work/345117#345117
+// Thanks for ledvinap for making such accuracy possible! See: https://github.com/cleanflight/cleanflight/issues/940#issuecomment-110323384
+// https://github.com/Crashpilot1000/HarakiriWebstore1/blob/master/src/mw.c#L1235
+#ifdef FAST_TRIGONOMETRY
+#define SINmagik1  1.0f
+#define SINmagik2 -1.666665710e-1f                                          // Double: -1.666665709650470145824129400050267289858e-1
+#define SINmagik3  8.333017292e-3f                                          // Double:  8.333017291562218127986291618761571373087e-3
+#define SINmagik4 -1.980661520e-4f                                          // Double: -1.980661520135080504411629636078917643846e-4
+#define SINmagik5  2.600054768e-6f                                          // Double:  2.600054767890361277123254766503271638682e-6
+float sin_approx(float x)
+{
+    int32_t xint = x;
+    if (xint < -32 || xint > 32) return 0.0f;                               // Stop here on error input (5 * 360 Deg)
+    while (x >  M_PIf) x -= (2.0f * M_PIf);                                 // always wrap input angle to -PI..PI
+    while (x < -M_PIf) x += (2.0f * M_PIf);
+    if      (x >  (0.5f * M_PIf)) x =  (0.5f * M_PIf) - (x - (0.5f * M_PIf));   // We just pick -90..+90 Degree
+    else if (x < -(0.5f * M_PIf)) x = -(0.5f * M_PIf) - ((0.5f * M_PIf) + x);
+    float x2 = x * x;
+    return x * (SINmagik1 + x2 * (SINmagik2 + x2 * (SINmagik3 + x2 * (SINmagik4 + x2 * SINmagik5))));
```

Comment:
Actually after unfolding 1 multiply I was able to go as low as looptime=950 on Harakiri PIDC in LEVEL mode. Looks impressive.